code/modules/wiremod/core/component.dm

```diff
@@ -266,6 +271,15 @@
 
 	return TRUE
 
+/// Called when trying to get the physical location of this object
+/obj/item/circuit_component/proc/get_location()
+	var/turf/current_turf = get_turf(src)
+	if(current_turf)
+		return current_turf
+	if(parent?.shell)
+		return get_turf(parent?.shell)
+	return null
```

Comment:
```suggestion
	return get_turf(parent?.shell)
```

Will just return null

This entire proc could just be `return get_turf(src) || get_turf(parent?.shell)` if you thought that was readable